### PR TITLE
init lesson 14 async

### PR DESCRIPTION
**Pull request by BTBTravis**
Hey Team here is what I got so far on Async coding for tonight lecture / workshop.

### CODE REVIEW

File: lessons/14_async/index.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+- Add things to your shopping cart
+- Check if a product is in stock
+- Send an Email
+- ...
+
+## What is async programming?
+
+Most code is synchronous. It is executed line by line until it is done. Sort of like reading a book start to finish. Asynchronous programming as you may have guessed is why code is executed in an out of order fashion. Like reading a book by starting on the 45th page and reading for 5 pages. We need async coding for things like calling an api because the request might take a while and we don't want our code to pause and wait for the response. The user might interact with the page in that time and we need the code to also respond to that. 
+
+In javascript there are three many ways of async programming. Through callbacks,, promises (bit older method) and via async functions (native support in modern browsers).
```
Comment (MrTim) [inline lessons/14_async/index.md]: probably you mean just three. Many not needed.

File: lessons/14_async/index.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+- Send an Email
+- ...
+
+## What is async programming?
+
+Most code is synchronous. It is executed line by line until it is done. Sort of like reading a book start to finish. Asynchronous programming as you may have guessed is why code is executed in an out of order fashion. Like reading a book by starting on the 45th page and reading for 5 pages. We need async coding for things like calling an api because the request might take a while and we don't want our code to pause and wait for the response. The user might interact with the page in that time and we need the code to also respond to that. 
+
+In javascript there are three many ways of async programming. Through callbacks,, promises (bit older method) and via async functions (native support in modern browsers). 
+
+Calls backs we won't cover in this lecture because we have done so extensively in pervious classes. Think about:
```
Comment (MrTim) [inline lessons/14_async/index.md]: Callbacks

**Review by MrTim — COMMENTED** [review]
Nice start!
I don't feel that this explains anything though(...Maybe you can put more strait forward examples with one `then` statement for promise and for await/async something similar. 
This is just a suggestion, since I think you would explain a lot in words and won't write it all down here.

File: lessons/14_async/index.md [hunk 3]
```diff
@@ -112,4 +112,32 @@ Async function with error handling
...
+## Fetch
+
+Resource Links:
+
+- [Docs](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch)
+- [Tutorial](https://javascript.info/fetch)
+
+## Classwork
+
+Lets go directly to playing with fetch(). Often times when working with APIs we need to call serval in order to preform some sort of operation like logging in. Today we are going to do something similar with a multi step math workflow. You are free to use promises, async functions, and callbacks as you with.
```
Comment (MrTim) [inline lessons/14_async/index.md:144]: call server

File: lessons/14_async/index.md [hunk 4]
```diff
@@ -112,4 +112,32 @@ Async function with error handling
...
+
+1. Start by putting some status text on the page ex: **loading...**
+2. Get three random numbers from `/random` endpoint with a GET request
```
Comment (MrTim) [inline lessons/14_async/index.md:147]: where those endpoints are pointing?

Comment (BTBTravis) [reply lessons/14_async/index.md:147]: added full documentation of classwork including full urls and instructions

**Comment by BTBTravis** [comment]
Thanks @MrTim for the comments. I updated most of the stuff. Should be a fun class!

**Pull request merged**